upload/library/SV/ReportImprovements/XenForo/ControllerPublic/Report.php

```diff
@@ -228,6 +234,79 @@ public function actionUpdate()
         return parent::actionUpdate();
     }
 
+    public function actionConversationJoin()
+    {
+        $reportId = $this->_input->filterSingle(
+            'report_id',
+            XenForo_Input::UINT
+        );
+        $report = $this->_getVisibleReportOrError($reportId);
+
+        if ($report['content_type'] !== 'conversation_message') {
+            return $this->responseError(
+                new XenForo_Phrase('requested_page_not_found'),
+                404
+            );
+        }
+
+        $conversation = $report['extraContent']['conversation'];
+
+        /** @var XenForo_Model_Conversation $conversationModel */
+        $conversationModel = $this->getModelFromCache(
+            'XenForo_Model_Conversation'
+        );
+
+        $conversationMaster = $conversationModel->getConversationMasterById(
+            $conversation['conversation_id']
+        );
+
+        if (!$conversationMaster) {
+            return $this->responseError(
+                new XenForo_Phrase('requested_conversation_not_found'),
+                404
+            );
+        }
+
+        $visitor = XenForo_Visitor::getInstance();
+
+        if (!$visitor->hasPermission('conversation', 'joinReported')) {
+            return $this->responseNoPermission();
+        }
+
+        if ($this->isConfirmedPost()) {
+            /** @var XenForo_DataWriter_ConversationMaster $conversationDw */
+            $conversationDw = XenForo_DataWriter::create(
+                'XenForo_DataWriter_ConversationMaster'
+            );
+            $conversationDw->setExistingData($conversationMaster, true);
+            $conversationDw->addRecipientUserIds(array($visitor['user_id']));
+            $conversationDw->save();
+
+            return $this->responseRedirect(
+                XenForo_ControllerResponse_Redirect::SUCCESS,
+                XenForo_Link::buildPublicLink(
+                    'conversations/message',
+                    array(
```

Comment:
Was peaking at the built-in conversation message report handler at the time, there's something similar in there for whatever reason. Fixed.